src/map_item_stack.h

```diff
@@ -43,8 +44,8 @@ class map_item_stack
 
 std::vector<map_item_stack> filter_item_stacks( const std::vector<map_item_stack> &stack,
         const std::string &filter );
-int list_filter_high_priority( std::vector<map_item_stack> &stack, const std::string &priorities );
+int list_filter_high_priority( std::vector<map_item_stack> &stack, const std::unordered_set<itype_id> *priorities );
 int list_filter_low_priority( std::vector<map_item_stack> &stack, int start,
-                              const std::string &priorities );
+                              const std::unordered_set<itype_id> *priorities );
```

Comment:
You're treating priorities as though it can never be null in the function definitions, so if they can be null, they should be safer (I didn't crash, so it's safe from at least casual examination), and if they can't, it just pass these as references.